pcs_scraper/scraper.py

```diff
@@ -7,39 +7,51 @@
 
 
 class Scraper:
-    def __init__(self):
-        self.driver = webdriver.Chrome()  # or any other Selenium driver
+    def __init__(self, headless=True):
+        if headless:
+            options = webdriver.ChromeOptions()
+            options.add_argument("--headless")
+            self.driver = webdriver.Chrome(options=options)
+        else:
+            self.driver = webdriver.Chrome()
+
         self.md_contents = {}
+        self.is_signed_in = False
 
     def scrape(self, url, user, password):
         """
         Given a repository URL, handles signing into GitHub and extracts
         the text contents of all markdown files in the repository.
         """
-
-        # try:
-        #     self.driver.get(url)
-        #     self.__sign_in(user, password)
-
-        #     for md_url in self.__fetch_markdowns():
-        #         self.md_contents[md_url] = self.__read_md(md_url)
-
-        # except TimeoutException:
-
         self.driver.get(url)
-        self.__sign_in(user, password)
+        if not self.is_signed_in:
+            self.__sign_in(user, password)
```

Comment:
If the user is not signed in. We should stop, rather than automatically signing in.

We may not have the username and password stored; and considering we **don't** want to store them, we can either prompt them to login or we simply stop.

Reply:
We will have the username and password stored. Signing in is necessary since we wouldn't be able to access the search results page otherwise. The plan is to let the user clone the repository and store their username and password in a .env file, the program then automatically pulls these details from the environment and uses them to sign in. This check is here so that the scraper doesn't try to sign in multiple times, since after vising the 1st repository the user will be already signed in and won't need to sign in again for subsequent repository during the crawling.

Reply:
Wasn't the original plan to **not** store those details? Is that not why we moved over to selenium? 

If *we are* storing them now, can't we use GitHub REST API or GraphQL to do the scarping instead.

Reply:
> Wasn't the original plan to not store those details? Is that not why we moved over to selenium?

We moved to selenium because we realised that we needed to store the details, not to avoid storing them. Initially, we would recursively explore a repository by accessing subfolders and what not, but due to the sheer number of requests you had to make (because large repositories have many subfolders) this turned out to be very inefficient. At this point we decided to use the search results page and log the user in using his details, and for this we needed selenium to manipulate the DOM to input these details.

> If we are storing them now, can't we use GitHub REST API or GraphQL to do the scarping instead.

Maybe, but this is a learning project and I think there's more value in learning scraping with selenium than using an API.